ocaml/xapi/xapi_vm_snapshot.ml

```diff
@@ -289,9 +299,19 @@ let update_vifs_vbds_and_vgpus ~__context ~snapshot ~vm =
 
 		debug "Cloning the snapshoted disks";
 		let driver_params = Xapi_vm_clone.make_driver_params () in
-		let cloned_disks = Xapi_vm_clone.safe_clone_disks rpc session_id Xapi_vm_clone.Disk_op_clone ~__context snap_vbds driver_params in
+		let cloned_disks = Xapi_vm_clone.safe_clone_disks rpc session_id Xapi_vm_clone.Disk_op_clone ~__context snap_vbds_without_cd driver_params in
 		TaskHelper.set_progress ~__context 0.5;
 
+		debug "Updating the snapshot_of fields for relevant VDIs";
+		let original_to_newvdi_map = List.map2 (fun oldvdi (_, newvdi, _) -> (oldvdi,newvdi)) vdis_snap_of cloned_disks in
+		List.iter (fun (vdi, snaps) ->
+			let newvdi = List.assoc vdi original_to_newvdi_map in
+			List.iter (fun s -> Db.VDI.set_snapshot_of ~__context ~self:s ~value:newvdi) snaps
+		) vdi_to_snapshots_map;
+
+		debug "Updating the snapshot_of fields for snapshot VDIs";
+		List.iter2 (fun snapvdi (_, newvdi, _) -> Db.VDI.set_snapshot_of ~__context ~self:snapvdi ~value:newvdi ) snap_vdis cloned_disks;
```

Comment:
Sorry if I've misunderstood, but why are these two lines needed? Haven't we already updated the `snapshot_of` field of each snapshot of each VDI at this point?